packages/react-native/src/posthog-rn.ts

```diff
@@ -816,6 +921,20 @@ export class PostHog extends PostHogCore {
     const previousDistinctId = this.getDistinctId()
     super.identify(distinctId, properties, options)
 
+    // Automatically cache person properties for feature flag evaluation
+    if (properties && Object.keys(properties).length > 0) {
+      const propsToCache: Record<string, string> = {}
+      Object.keys(properties).forEach((key) => {
+        const value = properties[key]
+        if (value !== null && value !== undefined) {
+          propsToCache[key] = String(value)
+        }
+      })
+      if (Object.keys(propsToCache).length > 0) {
+        this.setPersonPropertiesForFlags(propsToCache)
```

Comment:
similar to https://github.com/PostHog/posthog-js/pull/2564/files#r2518097170 since flags will be reloaded already during `identify` if the user wasnt identified before